Since CGAL 5.5 Modifiable_priority_queue compilation fails for `CGAL::CGAL_BOOST_PENDING_RELAXED_HEAP` in debug mode

## Issue Details

With #6155 (CGAL 5.5), if you modify `STL_Extension/test/STL_Extension/test_Modifiable_priority_queue.cpp` with the following type:
```cpp
typedef CGAL::Modifiable_priority_queue<Type*,Less,First_of_pair, CGAL::CGAL_BOOST_PENDING_RELAXED_HEAP> Queue;
```

the compilation fails in debug mode:
```txt
miniconda3/envs/gudhi-dev/include/CGAL/STL_Extension/internal/boost/relaxed_heap.hpp: In instantiation of ‘void CGAL::internal::boost_::relaxed_heap<IndexedType, Compare, ID>::remove(const value_type&) [with IndexedType = std::pair<long unsigned int, int>*; Compare = Less; ID = First_of_pair; CGAL::internal::boost_::relaxed_heap<IndexedType, Compare, ID>::value_type = std::pair<long unsigned int, int>*]’:
miniconda3/envs/gudhi-dev/include/CGAL/Modifiable_priority_queue.h:59:53:   required from ‘void CGAL::Modifiable_priority_queue<IndexedType_, Compare_, ID_, heap_type>::erase(const value_type&) [with IndexedType_ = std::pair<long unsigned int, int>*; Compare_ = Less; ID_ = First_of_pair; CGAL::Heap_type heap_type = CGAL::CGAL_BOOST_PENDING_RELAXED_HEAP; CGAL::Modifiable_priority_queue<IndexedType_, Compare_, ID_, heap_type>::value_type = std::pair<long unsigned int, int>*]’
STL_Extension/test/STL_Extension/test_Modifiable_priority_queue.cpp:75:10:   required from here
miniconda3/envs/gudhi-dev/include/CGAL/STL_Extension/internal/boost/relaxed_heap.hpp:197:9: error: no matching function for call to ‘possibly(__gnu_cxx::__alloc_traits<std::allocator<boost::optional<std::pair<long unsigned int, int>*> >, boost::optional<std::pair<long unsigned int, int>*> >::value_type&)’
  197 |         CGAL_assertion(groups[get(id, x)]);
      |         ^~~~~~~~~~~~~~
In file included from miniconda3/envs/gudhi-dev/include/CGAL/assertions.h:331,
                 from STL_Extension/test/STL_Extension/test_Modifiable_priority_queue.cpp:1:
miniconda3/envs/gudhi-dev/include/CGAL/Uncertain.h:253:13: note: candidate: ‘bool CGAL::possibly(bool)’
  253 | inline bool possibly(bool b) { return b; }
      |             ^~~~~~~~
miniconda3/envs/gudhi-dev/include/CGAL/Uncertain.h:253:27: note:   no known conversion for argument 1 from ‘__gnu_cxx::__alloc_traits<std::allocator<boost::optional<std::pair<long unsigned int, int>*> >, boost::optional<std::pair<long unsigned int, int>*> >::value_type’ {aka ‘boost::optional<std::pair<long unsigned int, int>*>’} to ‘bool’
  253 | inline bool possibly(bool b) { return b; }
      |                      ~~~~~^
miniconda3/envs/gudhi-dev/include/CGAL/Uncertain.h:265:6: note: candidate: ‘bool CGAL::possibly(CGAL::Uncertain<bool>)’
  265 | bool possibly(Uncertain<bool> c)
      |      ^~~~~~~~
miniconda3/envs/gudhi-dev/include/CGAL/Uncertain.h:265:31: note:   no known conversion for argument 1 from ‘__gnu_cxx::__alloc_traits<std::allocator<boost::optional<std::pair<long unsigned int, int>*> >, boost::optional<std::pair<long unsigned int, int>*> >::value_type’ {aka ‘boost::optional<std::pair<long unsigned int, int>*>’} to ‘CGAL::Uncertain<bool>’
  265 | bool possibly(Uncertain<bool> c)
      |               ~~~~~~~~~~~~~~~~^
```

## Source Code

[test_Modifiable_priority_queue.cpp](https://github.com/CGAL/cgal/blob/master/STL_Extension/test/STL_Extension/test_Modifiable_priority_queue.cpp)
with the following type:
```cpp
typedef CGAL::Modifiable_priority_queue<Type*,Less,First_of_pair, CGAL::CGAL_BOOST_PENDING_RELAXED_HEAP> Queue;
```

_In a private message, I would be interested to understand the difference between the [heap types](https://github.com/CGAL/cgal/blob/098ca52aff847697ab0d508baba438f54aaea546/STL_Extension/include/CGAL/Modifiable_priority_queue.h#L27) of Modifiable_priority_queue. Maybe we do not use it in a proper way._

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Linux 64
* Compiler: g++ 11.2.0
* Release or debug mode: Debug
* Specific flags used (if any):
* CGAL version: 5.5
* Boost version: 1.74.0
* Other libraries versions if used (Eigen, TBB, etc.):

--- a/STL_Extension/include/CGAL/STL_Extension/internal/boost/relaxed_heap.hpp
+++ b/STL_Extension/include/CGAL/STL_Extension/internal/boost/relaxed_heap.hpp
@@ -194,7 +194,7 @@ public:
     void remove(const value_type& x)
     {
         group* a = &index_to_group[get(id, x) / log_n];
-        CGAL_assertion(groups[get(id, x)]);
+        CGAL_assertion(static_cast< bool >(groups[get(id, x)]));
         a->value = x;
         a->kind = smallest_key;
         promote(a);

--- a/Surface_mesh_simplification/test/Surface_mesh_simplification/edge_collapse_topology.cpp
+++ b/Surface_mesh_simplification/test/Surface_mesh_simplification/edge_collapse_topology.cpp
@@ -43,6 +43,7 @@ int main(int argc, char** argv)
             ,stop
              ,CGAL::parameters::vertex_index_map(get(CGAL::vertex_external_index,surface))
                                .halfedge_index_map  (get(CGAL::halfedge_external_index  ,surface))
+                               .use_relaxed_order(CGAL::Tag_true())
            );
 
   std::cout << "\nFinished...\n" << r << " edges removed.\n"

